app.js

```diff
@@ -38,9 +38,103 @@ function operate(operator, a, b) {
 const display = document.querySelector(".display");
 const btns = document.querySelectorAll("button");
 
-// put number selected on the display
 btns.forEach((button) => {
-  button.addEventListener("click", (e) => {
-    display.innerHTML += button.innerText;
-  });
+  button.addEventListener("click", (e) => calculatorDisplay(button.id));
 });
+
+let firstNum = "";
+let secondNum = "";
+let operator;
+// put number selected on the display
+function calculatorDisplay(value) {
+  function getData(value) {
+    if (
+      display.innerHTML.includes("-") ||
+      display.innerHTML.includes("+") ||
+      display.innerHTML.includes("x") ||
+      display.innerHTML.includes("÷")
+    ) {
+      secondNum += value;
+    } else {
+      firstNum += value;
+    }
+  }
+
+  switch (value) {
+    case "zero":
+      display.innerHTML += "0";
+      getData("0");
+      break;
+    case "one":
+      display.innerHTML += "1";
+      getData("1");
+      break;
+    case "two":
+      display.innerHTML += "2";
+      getData("2");
+      break;
+    case "three":
+      display.innerHTML += "3";
+      getData("3");
+      break;
+    case "four":
+      display.innerHTML += "4";
+      getData("4");
+      break;
+    case "five":
+      display.innerHTML += "5";
+      getData("5");
+      break;
+    case "six":
+      display.innerHTML += "6";
+      getData("6");
+      break;
+    case "seven":
+      display.innerHTML += "7";
+      getData("7");
+      break;
+    case "eight":
+      display.innerHTML += "8";
+      getData("8");
+      break;
+    case "nine":
+      display.innerHTML += "9";
+      getData("9");
+      break;
+    case "decimal":
+      display.innerHTML += ".";
+      getData(".");
+      break;
```

Comment:
All of these cases do pretty much the same thing. If instead of passing a word (one, two, ...) you passed directly the number (0, 1, 2), you can collapse all of these cases into a single if statement:

```js
if (/^[0-9.]$/.match(value)) {
  display.innerHTML += value;
  getData(value);
} else {
 switch...
```

that is 44 lines turned into 4. The best code is no code: less code is easier to read, understand and maintain.